### DISCUSSION TITLE
Cpp.files.exclude

### DISCUSSION THREAD

**Discussion by jalanh11640** [microsoft/vscode-cpptools#11708] [Q&A]
Using c/cpp extension 1.18.5. 

I have the following in my user settings.json file.

    "C_Cpp.files.exclude": {
        "/usr/local/WindRiver/vxworks-7/vsb_rad750_3U_vx7_20201013/**": true
    }

Yet, when I have logging level set to Debug, I see the following in the log.

  tag parsing file: /usr/local/WindRiver/vxworks-7/vsb_rad750_3U_vx7_20201013/krnl/h/public/version.h

Shouldn't this setup prevent files in that directory from being parsed? If not, is there some setting that will prevent them from being parsed?

Thanks

**Comment by sean-mcmanus** [comment]
@jalanh11640 File-based exclusions are disabled by default due to performance reasons. Removing the `**` from the end should fix it (to exclude the entire folder). I've filed an issue at https://github.com/microsoft/vscode-cpptools/issues/11709 to track automatically doing that. The other alternative is to set C_Cpp.exclusionPolicy to "checkFilesAndFolders", but that should only be done if you actually need per-file exclusions.

**Reply by jalanh11640** [reply]
Awesome!  Entire folder exclusion is what I was looking for.

Thank you!

For anyone that stumbles upon this later, it appears the directories need the trailing '/'
